File: kotlin-ide/j2k/new/testData/newJ2k/assignmentExpression/declarationAssignmentsChain.java
class Test {
    public static void main(String[] args) {
        int a = 0;
        int b = 0;
        int c = 0;
        int d = 0;
        int e = 0;
        //-----
        int f = a = b += c = d *= e;
        //-----
        int g = a;
        //-----
        int h = a = b;
        //-----
        int i = a += b;
        //-----
        int j = a += b = c;
    }
}

File: nj2k/testData/newJ2k/assignmentExpression/declarationAssignmentsChain.kt
internal object Test {
    @JvmStatic
    fun main(args: Array<String>) {
        var a = 0
        var b = 0
        var c = 0
        var d = 0
        val e = 0
        //-----
        d *= e
        c = d
        b += c
        a = b
        val f = a
        //-----
        val g = a
        //-----
        a = b
        val h = a
        //-----
        a += b
        val i = a
        //-----
        b = c
        a += b
        val j = a
    }
}